guides/release/components/helper-functions.md

```diff
@@ -315,7 +315,69 @@ components.
 ```
 
 In the component's template, you can then use the `person` object:
-
 ```handlebars {data-filename=app/components/greeting/template.hbs}
 Hello, {{@person.givenName}} {{@person.familyName}}
 ```
+
+### The `in-element` helper
+
+Using the [`{{in-element}}`](https://api.emberjs.com/ember/3.27/classes/Ember.Templates.helpers/methods/in-element?anchor=in-element) helper, you can render content into a specific DOM element that is in a different part of the page. For instance, we might want
+to render a modal, tooltip or dropdown.
+
+Let's say I wanted to show a dropdown when a button is clicked on.
+
+Below, we have the button we want to click on, a div element that will hold the dropdown and the dropdown component. We pass in `showDropdown` that will add the dropdown into our div.
+```handlebars {data-filename=app/components/some-component.hbs}
+  <button type="button" {{on "click" this.onClickShowDropdown}}>More Actions</button>
+  <div id="dropdown-destination" />
+
+  <MyDropdownComponent
+    @show={{this.showDropdown}}
+  />
+```
+
+When the user clicks on the button, the flag `showDropdown` will be set to `true`.
+```js {data-filename=app/components/some-component.js}
+  @tracked
+  showDropdown = false;
+
+  @action
+  onClickShowDropdown() {
+    this.showDropdown = true;
+  }
+```
+
+Since `showDropdown` is passed into `MyDropdownComponent` as `show`, when `show` is `true`, the `in-element` helper is activated. We pass in `destinationElement` into `in-element` which contains the destination DOM element. The `in-element` helper will take the content block and render into the destination element.
+```handlebars {data-filename=app/components/my-dropdown-component.hbs}
+{{#if @show}}
+  {{#in-element this.destinationElement}}
+    <ul>
+      <li>Archive</li>
+      <li>Mark as Read</li>
+      <li>Report</li>
+    </ul>
+  {{/in-element}}
+{{/if}}
+```
+
+```js {data-filename=app/components/my-dropdown-component.js}
+  get destinationElement() {
+    return document.querySelector('#dropdown-destination');
+  }
+```
+
+After the user clicks on the button, the final HTML result for the div will be like this:
+```html
+  <div id="dropdown-destination">
+    <ul>
+      <li>Archive</li>
+      <li>Mark as Read</li>
+      <li>Report</li>
+    </ul>
+  </div>
+```
+
+Things to note:
+- The destination element needs to exist in the DOM before we use the helper. If not, an error will be thrown in development mode, but not in production.
+- When the destination element changes, the contents defined in `in-element` will re-render completely.
+- By default, the `in-element` helper will replace the entire contents of the destination element. If we want it to just append, we need to pass `insertBefore=null`.
```

Comment:
I found the 3rd point confusing (the API documentation wasn't as clear as it could be either). I think what you and the documentation meant to say is the following. Can you check?

```suggestion
- The destination element needs to exist in the DOM before we use the helper. Otherwise, an error will be thrown if you are in development mode. The error is not thrown in production.
- When the destination element changes, the content defined in `in-element` will re-render completely.
- By default, the `in-element` helper replaces the destination element's existing content with the helper's block content. If you want to instead append the block content, you can pass `insertBefore=null`.
```

Reply:
Yup!